src/codec.rs

```diff
@@ -756,19 +822,11 @@ impl_array!(
 	253, 254, 255, 256, 384, 512, 768, 1024, 2048, 4096, 8192, 16384, 32768,
 );
 
-impl<T: Encode> Encode for Box<T> {
-	fn encode_to<W: Output>(&self, dest: &mut W) {
-		self.as_ref().encode_to(dest)
-	}
-}
-
-impl<T: Decode> Decode for Box<T> {
-	fn decode<I: Input>(input: &mut I) -> Result<Self, Error> {
-		Ok(Box::new(T::decode(input)?))
+impl Encode for [u8] {
+	fn size_hint(&self) -> usize {
+		self.len() + mem::size_of::<u32>()
```

Comment:
That won't really work, since `size_hint` is not implemented for `Compact/CompactRef`.
I've only implemented it for objects that can potentially contain large (kBytes, MBytes) values that would benefit from this optimization to avoid cluttering the code.
Note it's also perfectly fine to overestimate a bit, the actual allocation is likely to be larger than this value anyway.

Reply:
Yeah, makes sense :)